dask/array/random.py

```diff
@@ -17,7 +17,603 @@
 from dask.array.creation import arange
 from dask.base import tokenize
 from dask.highlevelgraph import HighLevelGraph
-from dask.utils import derived_from, random_state_data
+from dask.utils import derived_from, random_rng_data, random_state_data
+
+
+class Generator:
+    """
+    Container for the BitGenerators.
+
+    ``Generator`` exposes a number of methods for generating random
+    numbers drawn from a variety of probability distributions. In addition to
+    the distribution-specific arguments, each method takes a keyword argument
+    `size` that defaults to ``None``. If `size` is ``None``, then a single
+    value is generated and returned. If `size` is an integer, then a 1-D
+    array filled with generated values is returned. If `size` is a tuple,
+    then an array with that shape is filled and returned.
+
+    The function :func:`dask.array.random.default_rng` will instantiate
+    a `Generator` with numpy's default `BitGenerator`.
+
+    **No Compatibility Guarantee**
+
+    ``Generator`` does not provide a version compatibility guarantee. In
+    particular, as better algorithms evolve the bit stream may change.
+
+    This object is identical to ``np.random.Generator`` except that all
+    functions also take a ``chunks=`` keyword argument.
+
+    Parameters
+    ----------
+    bit_generator : BitGenerator
+        BitGenerator to use as the core generator.
+
+    Notes
+    -----
+    The Python stdlib module `random` contains pseudo-random number generator
+    with a number of methods that are similar to the ones available in
+    ``Generator``. It uses Mersenne Twister, and this bit generator can
+    be accessed using ``MT19937``. ``Generator``, besides being
+    Dask-aware, has the advantage that it provides a much larger number
+    of probability distributions to choose from.
+
+    Examples
+    --------
+    >>> from numpy.random import PCG64
+    >>> from dask.array.random import Generator
+    >>> rng = Generator(PCG64())
+    >>> rng.standard_normal().compute() # doctest: +SKIP
+    array(0.44595957)  # random
+
+    See Also
+    --------
+    default_rng : Recommended constructor for `Generator`.
+    np.random.Generator
+    """
+
+    def __init__(self, bit_generator):
+        self._bit_generator = bit_generator
+        self._bit_generator_state = bit_generator.state
```

Comment:
Since this is already defined on the object, do we need to store it separately or can we just use that attribute?

Reply:
Now that we have decided to support numpy-like API only, there is indeed no need to store it separately.  Removed.